### PR TITLE
replace Road and Anchor with Edge (inheriting from RoadView)

### PR DESCRIPTION
**Pull request by funketh**
* Edge is used instead of Anchor but replaces Road as well
* Edges can only exist as pairs of 2 (one per Vertex) and represent a road
    * data about the road is shared between the 2 Edge objects
* an Edge represents how the road would look like if you were to stand at one of its endpoints (there is one Edge object for each endpoint)
* Since roads are constructed before vertices, there are RoadView objects which are just like Edge objects but they don't have a Vertex assigned to them yet
* Point has now become redundant; Vector2D will be used instead

### CODE REVIEW

**Review by RichardKarlFuchs — APPROVED** [review]

File: Assets/Scripts/DataTypes/Car.cs [hunk 1]
```diff
@@ -3,13 +3,13 @@ namespace DataTypes
     public class Car
     {
         public readonly int id;
-        public Road road;
+        public Edge road;
         public float positionOnRoad;
```
Comment (Shafma) [inline Assets/Scripts/DataTypes/Car.cs:7]: Wie genau kann ich mir das vorstellen? Eine Prozentangabe, Distanz etc.

**Comment by rrickfox** [comment]
Die Anzahl an Distance_Units, die das Auto zurückgelegt hat

Shafma <notifications@github.com> schrieb am Sa., 25. Jan. 2020, 16:35:

File: Assets/Scripts/DataTypes/Car.cs (hunk 1 above)
Comment (RichardKarlFuchs) [reply Assets/Scripts/DataTypes/Car.cs:7]: Wenn du positionOnRoad meinst, handelt es sich um die Anzahl an Distance Units, die das Auto zurückgelegt hat

**Review by Shafma — COMMENTED** [review]
Mir ist gerade nicht klar, was uns die Inheritance von Vertex überhaupt noch bringt.

**Comment by funketh** [comment]
In der Vertex Klasse wird die pathfinding Methode implementiert sein:
```cs
public Path FindPath(Vertex endVertex)
{
    // find path by iterating over self._edges
}
```
Damit wir halt nicht für jede Section nen eigenes pathfinding implementieren müssen, haben wir den Array an Edges.

**Review by Shafma — APPROVED** [review]

**Pull request merged**